controllers/validation.go

```diff
@@ -121,3 +122,41 @@ func (r *ClusterGroupUpgradeReconciler) validateOpenshiftUpgradeVersion(
 
 	return nil
 }
+
+func indexOf(element string, data []string) (int, error) {
+	for k, v := range data {
+		if element == v {
+			return k, nil
+		}
+	}
+	return -1, errors.New("element not found in data")
+}
+
+func (r *ClusterGroupUpgradeReconciler) validatePoliciesDependenciesOrder(
```

Comment:
Why don't we automatically order managed policies to satisfy dependencies?

Reply:
@jc-rh what do you think about this?
It makes sense to me.

Reply:
What if there is a circular dependency? I would rather flag it and let user decide what to fix. We can discuss what is the best way to let user specify a policy set with built-in dependency. It seems to be cleaner to have a different spec field rather than switching the order in the existing list.

Reply:
I did consider circular dependencies and thought we could have a warning for that, to help the user. I think it's fine to leave it as is for now and keep this in mind for the future.

Reply:
Sure, feel free to start capturing potential use cases in https://github.com/jc-rh/kni-enhancements/blob/main/enhancements/talm-acm-alignment-policy-dependency-support.md
The scope of this change is to make sure TALM still functional with policies with dependency.